projects/plugins/jetpack/extensions/plugins/publicize/components/settings-button/index.js

```diff
@@ -2,64 +2,62 @@
  * Publicize settings button component.
  *
  * Component which allows user to click to open settings
- * in a new window/tab. If window/tab is closed, then
- * connections will be automatically refreshed.
+ * in a new window/tab.
+ * If window/tab is closed,
+ * then connections will be automatically refreshed.
  */
+import { debounce } from 'lodash';
+import PageVisibility from 'react-page-visibility';
 
 /**
- * External dependencies
+ * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import classnames from 'classnames';
 import { ExternalLink } from '@wordpress/components';
+import { useEffect } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import getSiteFragment from '../../../../shared/get-site-fragment';
+import useSelectSocialMediaConnections from '../../hooks/use-social-media-connections';
 
-export default function PublicizeSettingsButton( props ) {
-	function getButtonLink() {
-		const siteFragment = getSiteFragment();
+const refreshThreshold = 2000;
 
-		// If running in WP.com wp-admin or in Calypso, we redirect to Calypso sharing settings.
-		if ( siteFragment ) {
-			return `https://wordpress.com/marketing/connections/${ siteFragment }`;
-		}
+export default function PublicizeSettingsButton() {
+	const { refresh } = useSelectSocialMediaConnections();
+	const siteFragment = getSiteFragment();
+
+	// Refresh connections when mounted.
+	useEffect( () => {
+		refresh();
+	}, [ refresh ] );
```

Comment:
What does this fix? The connections should be in sync as the editor loads, so we shouldn't need to refresh them again.